libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorRecognizer.cs

```diff
@@ -317,53 +313,60 @@ private static JToken EntityResultToInstanceJObject(string text, Result result)
+            if (this._resolver == null)
             {
                 return;
             }
 
-            var results = _resolver.Score(text, LabelType.Entity);
-            recognizerResult.Properties.Add(EntitiesProperty, results);
-
-            if (results.Any())
+            try
             {
-                if (recognizerResult.Entities == null)
-                {
-                    recognizerResult.Entities = new JObject();
-                }
+                var results = _resolver.Score(text, LabelType.Entity);
+                recognizerResult.Properties.Add(EntitiesProperty, results);
 
-                var entitiesResult = recognizerResult.Entities;
-                foreach (var result in results)
+                if (results.Any())
                 {
-                    // add value
-                    JToken values;
-                    if (!entitiesResult.TryGetValue(result.Label.Name, StringComparison.OrdinalIgnoreCase, out values))
+                    if (recognizerResult.Entities == null)
                     {
-                        values = new JArray();
-                        entitiesResult[result.Label.Name] = values;
+                        recognizerResult.Entities = new JObject();
                     }
 
-                    ((JArray)values).Add(EntityResultToJObject(text, result));
-
-                    // get/create $instance
-                    JToken instanceRoot;
-                    if (!recognizerResult.Entities.TryGetValue("$instance", StringComparison.OrdinalIgnoreCase, out instanceRoot))
+                    var entitiesResult = recognizerResult.Entities;
+                    foreach (var result in results)
                     {
-                        instanceRoot = new JObject();
-                        recognizerResult.Entities["$instance"] = instanceRoot;
-                    }
+                        // add value
+                        JToken values;
+                        if (!entitiesResult.TryGetValue(result.Label.Name, StringComparison.OrdinalIgnoreCase, out values))
+                        {
+                            values = new JArray();
+                            entitiesResult[result.Label.Name] = values;
+                        }
 
-                    // add instanceData
-                    JToken instanceData;
-                    if (!((JObject)instanceRoot).TryGetValue(result.Label.Name, StringComparison.OrdinalIgnoreCase, out instanceData))
-                    {
-                        instanceData = new JArray();
-                        instanceRoot[result.Label.Name] = instanceData;
-                    }
+                        ((JArray)values).Add(EntityResultToJObject(text, result));
+
+                        // get/create $instance
+                        JToken instanceRoot;
+                        if (!recognizerResult.Entities.TryGetValue("$instance", StringComparison.OrdinalIgnoreCase, out instanceRoot))
```

Comment:
$instance is used in many places, consider adding a constant so we centralize the definition.

Reply:
Not sure what $instance is for. Will not touch it in this PR.